umh-core/pkg/service/redpanda/redpanda.go

```diff
@@ -0,0 +1,813 @@
+				if label == "produce" {
+					metrics.Throughput.BytesIn = int64(getValue(metric))
+				} else if label == "consume" {
+					metrics.Throughput.BytesOut = int64(getValue(metric))
+				}
+			}
+
+		}
+	}
+
+	return metrics, nil
+}
+
+// GetHealthCheckAndMetrics returns the health check and metrics of a Redpanda service
+func (s *RedpandaService) GetHealthCheckAndMetrics(ctx context.Context, tick uint64) (RedpandaStatus, error) {
+	start := time.Now()
+	defer func() {
+		metrics.ObserveReconcileTime(logger.ComponentRedpandaService, constants.RedpandaServiceName, time.Since(start))
+	}()
+
+	if ctx.Err() != nil {
+		return RedpandaStatus{}, ctx.Err()
+	}
+
+	// Skip health checks and metrics if the service doesn't exist yet
+	// This avoids unnecessary errors in Status() when the service is still being created
+	if _, exists := s.s6Manager.GetInstance(constants.RedpandaServiceName); !exists {
+		return RedpandaStatus{
+			HealthCheck: HealthCheck{
+				IsLive:  false,
+				IsReady: false,
+			},
+			Metrics: Metrics{},
+			Logs:    []s6service.LogEntry{},
+		}, nil
+	}
+
+	baseURL := "http://localhost:9644/"
+	metricsEndpoint := "public_metrics"
+	livenessEndpoint := "ping"
+	readinessEndpoint := "ready"
+
+	// Helper function to make HTTP requests with context
+	doRequest := func(endpoint string) (*http.Response, error) {
+		start := time.Now()
+
+		defer func() {
+			s.logger.Debugf("Request for %s took %s", endpoint, time.Since(start))
+		}()
+
+		req, err := http.NewRequestWithContext(ctx, http.MethodGet, baseURL+endpoint, nil)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create request for %s: %w", endpoint, err)
+		}
+		resp, err := s.httpClient.Do(req)
+		if err != nil {
+			return nil, fmt.Errorf("failed to execute request for %s: %w", endpoint, err)
+		}
+		return resp, nil
+	}
+
+	var healthCheck HealthCheck
+
+	// Check liveness
+	if resp, err := doRequest(livenessEndpoint); err == nil && resp != nil {
+		healthCheck.IsLive = resp.StatusCode == http.StatusOK
+		resp.Body.Close()
+	} else {
+		return RedpandaStatus{}, fmt.Errorf("failed to check liveness: %w", err)
+	}
+
+	// Check readiness
+	if resp, err := doRequest(readinessEndpoint); err == nil && resp != nil {
+		defer resp.Body.Close()
+
+		// Even if status is 503, we still want to read the body to get the detailed status
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return RedpandaStatus{}, fmt.Errorf("failed to read ready response body: %w", err)
+		}
+
+		// Service is ready if status is 200 and there's no error
+		healthCheck.IsReady = resp.StatusCode == http.StatusOK
+
+		// Log detailed status if not ready
+		if !healthCheck.IsReady {
+			s.logger.Debugw("Service not ready",
+				"service", constants.RedpandaServiceName,
+				"error", string(body))
+		}
+	} else {
+		return RedpandaStatus{}, fmt.Errorf("failed to check readiness: %w", err)
+	}
+
+	// Get version
+	healthCheck.Version = constants.RedpandaVersion
```

Comment:
ah now I get it why you define it. would probably either fetch it from the health endpoitn and then compare it with what is in the dockerfile

Reply:
Moved this.
Now we compile the version into the image.
Imo this is better then to do http requests over and over.